### ISSUE TITLE
geoopt.nn?

### ISSUE THREAD

**Issue by ferrine** [geoopt/geoopt#113]
Where should we place reusable deep learning layers? Should it be `geoopt.nn` or a distinct package? We can also consider `geoopt.experimental` indicating it is under extensive development and considered to be changed without any warning.

Things I am ready to add so far:

* `nn.ManifoldEmbedding`
* `nn.hyperbolic.MobiusLinear`
* `nn.hyperbolic.MobiusDist2Hyperplanes`
* `nn.hyperbolic.MobiusDist2Centroids`
* `nn.hyperbolic.Exp` (with probably learnable origin)
* `nn.hyperbolic.Log` (with probably learnable origin)

CC @rrkarim, @newkozlukov

**Comment by ferrine** [comment]
My point is a slight concern it should go under geoopt as it is very restrictive for development, better to create a side package.

**Comment by rrkarim** [comment]
I would say separating packages makes more sense based on the initial idea of this package. Also, I guess, it simplifies the work of users in a way of not caring about the base implementation of all background math, but, instead, concentrating on the modules (which you are planning to provide) of the final architecture. And, yeah, easier life for devs too.

**Comment by ferrine** [comment]
Well, I think We need to push that. I think we need to merge, actually. It will serve as a development guide

**Comment by newkozlukov** [comment]
A year later, I'd say single repo and `.nn` sounds OK-ish

I would be OK re-joining the effort somewhat after Nov 17

**Comment by ferrine** [comment]
Yeah, a year later, why not? I have open-source meetings on Sundays once in two weeks, we can meet there and start a reunion there

**Comment by newkozlukov** [comment]
Great, here's first-Sunday-after-Nov-17

<a target="_blank" href="https://calendar.google.com/event?action=TEMPLATE&amp;tmeid=NWxpcTVsbGU5Y3IyajI5Z285cXY0aTlhaHIgbmV3a296bHVrb3ZAbQ&amp;tmsrc=newkozlukov%40gmail.com"><img border="0" src="https://www.google.com/calendar/images/ext/gc_button1_en.gif"></a>
